hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java

```diff
@@ -302,8 +297,15 @@ public boolean allTokenBalancesVanish(AccountID aId) {
 			throw new IllegalStateException("Ledger has no manageable token relationships!");
 		}
 
-		var tokens = (MerkleAccountTokens) accountsLedger.get(aId, TOKENS);
-		for (TokenID tId : tokens.asTokenIds()) {
+		var currKey = (EntityNumPair) accountsLedger.get(aId, LAST_ASSOCIATED_TOKEN);
+		final List<TokenID> listOfAssociatedTokens = new ArrayList<>();
+		// get All the tokenIds associated by traversing the linkedList
+		while (!currKey.equals(MISSING_NUM_PAIR)) {
```

Comment:
Yikes, this means we also need to maintain a `hasNonZeroTokenBalances` in `MerkleAccountState`.

This would make `CryptoDelete` a trivial DOS vector---just create an account with 10k token associations, at least one of which has a non-zero token balance; and then spam the network with `CryptoDelete`s...each of which will **eventually** fail with `TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES`.